python/cudf/cudf/core/dataframe.py

```diff
@@ -1548,10 +1548,18 @@ def _concat(
                     cudf.core.index.as_index(out.index._values)
                 )
 
-        # Reassign precision for any decimal cols
+        # Reassign precision for decimal cols & type schema for struct cols
         for name, col in out._data.items():
-            if isinstance(col, cudf.core.column.Decimal64Column):
-                col = col._with_type_metadata(tables[0]._data[name].dtype)
+            if isinstance(
+                col,
+                (
+                    cudf.core.column.Decimal64Column,
+                    cudf.core.column.StructColumn,
+                ),
+            ):
+                out._data[name] = col._with_type_metadata(
+                    tables[0]._data[name].dtype
```

Comment:
One last comment: it looks like `Series._concat` handles the `Decimal64Column` and `StructColumn` differently, but this code path handles them the same way:
https://github.com/rapidsai/cudf/blob/0722e2027f6c80e494bda6b23dea98f80f0a4134/python/cudf/cudf/core/series.py#L1451-L1455

That seems suspicious to have very similar but not identical logic.

Reply:
It seems to be calling the same `_with_type_metadata` no?

Reply:
The arguments differ -- the `Series._concat` is calling `col._with_type_metadata(objs[0]._column.dtype)` for `Decimal64Column` and `col._with_type_metadata(objs[0].dtype)` for `StructColumn`.

In this `DataFrame._concat` function, both use `tables[0]._data[name].dtype` as the argument. Why does the Series implementation have an extra "`._column`"?

Reply:
Ah, I think it seems to be just a preference. `Series.dtype` is an alias for `Series._column.dtype`

Reply:
(I am struggling to reason about this code because the `Series._concat` method doesn't specify what type(s) `objs` can be -- so I'm unsure whether the extra `._column` is needed in that code.)

Reply:
@galipremsagar Okay great! I would like us to align the implementations between `Series._concat` and `DataFrame._concat` as closely as possible before merging, especially since we're using two aliases of the same attribute in `Series._concat`.

Reply:
Unified the two statements 👍